### DISCUSSION TITLE
.NET 6 Winforms Designer / Argument Exception: Illegal characters in path

### DISCUSSION THREAD

**Discussion by thomasmiko** [dotnet/winforms#6826] [Q&A]
My Issue was closed and converted in an discussion which cannot be opened (at least on my side), so i will open a new one :)

Issue: [https://github.com/dotnet/winforms/issues/6810](url)
Discussion (which i can't open) [https://github.com/dotnet/winforms/discussions/6818](url)

**Problem:**
I've have installed VS2022 (Latest Preview and latest stable) and in both versions when I edit a .NET 6 WinForms project and want to open the Designer for a UserControl / Form, the following exception appears

![2022_03_08_11_11_40_Window](https://user-images.githubusercontent.com/55090340/157630181-51cf9a2a-46b1-4eb1-b1a7-2fb617fc447a.png)

This is the output from Winforms Designer logging (Verbose)

_[11:06:01.0283033] Creating VSCodeDomDesignerLoader
[11:06:01.0543468] Processed 'System.Design' in 00:00:00.0998389
[11:06:01.0638662] Added 'System.Design'
[11:06:01.0648675] Loaded System.Drawing.Design in 00:00:00.0000189 [thread 35]
[11:06:01.0648675] Processed 'System.Drawing.Design' in 00:00:00.0001717
[11:06:01.0648675] Added 'System.Drawing.Design'
[11:06:01.2783607] [Thread 1] Creating design-time ITypeResolutionService.
[11:06:01.3429998] Current VS Version = 17.0
[11:06:01.3439981] Start processing enqueued assemblies
[11:06:01.3449974] Loaded Microsoft.WinForms.DesignTools.Protocol in 00:00:00.0000081 [thread 31]
[11:06:01.3599982] Processed 'Microsoft.WinForms.DesignTools.Protocol' in 00:00:00.0149551
[11:06:01.3619986] Added 'Microsoft.WinForms.DesignTools.Protocol'
[11:06:01.3619986] Loaded Microsoft.WinForms.DesignTools.Client in 00:00:00.0000074 [thread 31]
[11:06:01.3849983] Processed 'Microsoft.WinForms.DesignTools.Client' in 00:00:00.0224528
[11:06:01.3859974] Added 'Microsoft.WinForms.DesignTools.Client'
[11:06:01.5205867] [Thread 68] Launching server...
[11:06:03.3003038] System.ArgumentException: Illegal characters in path.
at System.Security.Permissions.FileIOPermission.EmulateFileIOPermissionChecks(String fullPath)
at System.IO.Directory.InternalCreateDirectoryHelper(String path, Boolean checkHost)
at Microsoft.WinForms.Utilities.Shared.ShadowCacheFolder..ctor(Boolean isNetCoreProject)
at Microsoft.DotNet.DesignTools.Client.Host.ServerLayout.d__8.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
at Microsoft.DotNet.DesignTools.Client.Host.ServerProcess.d__21.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
at Microsoft.DotNet.DesignTools.Client.Host.ServerHostFactory.d__7.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
at Microsoft.DotNet.DesignTools.Client.DesignToolsClientLoader.d__28.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
at Microsoft.VisualStudio.Telemetry.WindowsErrorReporting.WatsonReport.GetClrWatsonExceptionInfo(Exception exceptionObject)
[11:06:04.1210052] Cannot update selection in server process as Session is not connected._

I've tried it with several .NET 6 Projects, allways the same error.

Any ideas on how to proceed further to troubleshoot or further isolate the problem?

**Comment by RussKie** [comment]
Yes, sorry, something went wrong on the GitHub side, the GitHub support is looking at the issue. Thank you for re-raising it.

* Is it an existing project? If so, was it ported from .NET Framework?
* Did this form/control ever shown in the .NET out-of-process designer (i.e., targeting .NET and not .NET Framework)? If so, in what version of VS did it work?
* Is this issue specific to particular form/control - or - this project - or - affects any project?
* Does you form/control execute any "user" code in either .ctor or one of the overloads (e.g., `OnLoad()`)?


Process Monitor may also come handy investigating what path is causing the issue. If all else fails, can you provide a standalone repro?

Thank you

**Reply by thomasmiko** [reply]
Hi,

I can reproduce the problem on my computer with every new project I create that is based on .NET Core 3.1 / .NET 5 / .NET 6. The problem occurs the first time Form1.cs is opened, which is initially created using the wizard.

When I create a new .NET Framework 4.72 project, the designer opens correctly (even with existing .NET Framework 4.72 projects).

I've looked with Process Monitor for some special paths, but didn't see anything wrong. 
Is it somehow possible to attach the log file (*.PML) here? Or can I paste the CSV output from the Process Monitor?

Thanks and best Regards
Thomas

**Reply by RussKie** [reply]
.NET Framework is the original, in-process designer. .NET uses the [out-of-process designer](https://devblogs.microsoft.com/dotnet/state-of-the-windows-forms-designer-for-net-applications/).

What version of VS are you using? _Help > About Microsoft Visual Studio_

**Reply by thomasmiko** [reply]
I've installed two versions, i can reproduce the issue with both versions:

Microsoft Visual Studio Enterprise 2022 (64-bit) - Current
Version 17.1.1

Microsoft Visual Studio Enterprise 2022 (64-bit) - Preview
Version 17.2.0 Preview 1.0

**Reply by Shyam-Gupta** [reply]
@thomasmiko Would it be possible to confirm if this issue reproes on any other machine ? Also, could you please share .csproj file contents of a simple project in which this issue reproes.

When this issue reproes could you please connect a debugger to Visual Studio instance and share information about 'mscorlib.dll' loaded in the process. Steps:

1. Close all open instances of Visual Studio
2. Launch Visual Studio, say instance 1, and repro the issue.
3. Launch another instance of Visual Studio, say instance 2. Don't open any solution/project in it.
4. From instance 2, go to Debug -> Attach to Process
5. In 'Available Processes' list, select 'devenv.exe' and click Attach.
6. Now in instance 2, go to Debug -> Windows -> Modules
7. Look for mscorlib.dll, and share its Path, and Version

Thank you.

FYI: @merriemcgaw, @JeremyKuhne

**Reply by thomasmiko** [reply]
I can't reproduce the issue on another machine (tried it on my private machine and another machine of a collegue). 
The problem is currently only occurring on my work computer.

Here is the Screenshot of the loaded mscorlib.dll from the first devenv process

![2022_03_11_08_21_55_Source_Control_Explorer_Running_Microsoft_Visual_Studio_Preview](https://user-images.githubusercontent.com/55090340/157821385-18b4cc4c-c127-4d17-9f92-468100f26ae2.png)

And this is the content of the csproj file

![2022-03-11 08_26_57-](https://user-images.githubusercontent.com/55090340/157821796-f63ea5e4-a56b-4f80-b61a-138ab50c05eb.png)

**Reply by Shyam-Gupta** [reply]
Thank you for sharing the data. The project settings look fine and 'mscorlib.dll' is up to date.
This issue seems specific to the machine at present. So please bear with me as I come up with additional questions to gather more information about your environment. It will help me replicate the environment at my end and come up with a repro and root cause.

Could you please share list of extensions installed on your Visual Studio instance. Steps:
1. Go to Visual Studio -> Manage Extensions -> Installed -> All
2. Please share screenshot or their names

Share the components installed with Visual Studio. Steps:
1. Go to Visual Studio -> Help -> About Microsoft Visual Studio
2. Click 'Copy Info' and share with us

**Reply by thomasmiko** [reply]
Sorry for the delay, attached the required informations.

**Extensions**

![2022_03_14_17_36_04_Manage_Extensions](https://user-images.githubusercontent.com/55090340/158218711-ea361f7c-912d-4fdd-9953-3c6c18419288.png)
![2022_03_14_17_36_18_Manage_Extensions](https://user-images.githubusercontent.com/55090340/158218721-d983311e-a378-4f16-b344-16a798b30512.png)
![2022_03_14_17_36_33_Manage_Extensions](https://user-images.githubusercontent.com/55090340/158218738-2e49368c-9064-4ae1-8798-6dca4070dc22.png)

**Components**

Microsoft Visual Studio Enterprise 2022
Version 17.2.0 Preview 1.0
VisualStudio.17.Preview/17.2.0-pre.1.0+32210.308
Microsoft .NET Framework
Version 4.8.04084

Installed Version: Enterprise

Visual C++ 2022   00476-80000-00000-AA734
Microsoft Visual C++ 2022

.NET Core Debugging with WSL   1.0
.NET Core Debugging with WSL

ASP.NET and Web Tools 2019   17.2.140.2270
ASP.NET and Web Tools 2019

ASP.NET Web Frameworks and Tools 2019   17.2.140.2270
For additional information, visit https://www.asp.net/

Azure App Service Tools v3.0.0   17.2.140.2270
Azure App Service Tools v3.0.0

Azure Functions and Web Jobs Tools   17.2.140.2270
Azure Functions and Web Jobs Tools

C# Tools   4.2.0-1.22108.11+a29b6ae89a8885856e4c55a0e92ba056e1530458
C# components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Common Azure Tools   1.10
Provides common services for use by Azure Mobile Services and Microsoft Azure Tools.

ConvertToDevExtremeCommand Extension   1.0
ConvertToDevExtremeCommand Visual Studio Extension Detailed Info

CreateLayoutWizard   1.0
Create layout wizard.

DevExpress Reporting Extension   1.4
A Visual Studio extension that invokes the Report Designer editor for report definition VSREPX files.

DevExpress.DeploymentTool   1.0
A useful tool for deploying DevExpress assemblies.

DevExpress.Win.LayoutAssistant Extension   1.0
DevExpress.Win.LayoutAssistant Visual Studio Extension Detailed Info

Extensibility Message Bus   1.2.6 (master@34d6af2)
Provides common messaging-based MEF services for loosely coupled Visual Studio extension components communication and integration.

Microsoft Azure Tools for Visual Studio   2.9
Support for Azure Cloud Services projects

Microsoft JVM Debugger   1.0
Provides support for connecting the Visual Studio debugger to JDWP compatible Java Virtual Machines

Microsoft Library Manager   2.1.161+abc97ecc7d.RR
Install client-side libraries easily to any web project

Microsoft MI-Based Debugger   1.0
Provides support for connecting Visual Studio to MI compatible debuggers

Microsoft Visual C++ Wizards   1.0
Microsoft Visual C++ Wizards

Microsoft Visual Studio Tools for Containers   1.2
Develop, run, validate your ASP.NET Core applications in the target environment. F5 your application directly into a container with debugging, or CTRL + F5 to edit & refresh your app without having to rebuild the container.

Microsoft Visual Studio VC Package   1.0
Microsoft Visual Studio VC Package

Mono Debugging for Visual Studio   17.1.10 (65f98f5)
Support for debugging Mono processes with Visual Studio.

NuGet Package Manager   6.2.0
NuGet Package Manager in Visual Studio. For more information about NuGet, visit https://docs.nuget.org/

Office Developer Tools for Visual Studio   17.0.32009.00
Microsoft Office Developer Tools for Visual Studio

Razor (ASP.NET Core)   17.0.0.2210704+10268da7460a71ce637a4f1a92232f72b8e0528e
Provides languages services for ASP.NET Core Razor.

Snapshot Debugging Extension   1.0
Snapshot Debugging Visual Studio Extension Detailed Info

SQL Server Data Tools   17.0.62202.03110
Microsoft SQL Server Data Tools

Test Adapter for Boost.Test   1.0
Enables Visual Studio's testing tools with unit tests written for Boost.Test.  The use terms and Third Party Notices are available in the extension installation directory.

Test Adapter for Google Test   1.0
Enables Visual Studio's testing tools with unit tests written for Google Test.  The use terms and Third Party Notices are available in the extension installation directory.

TypeScript Tools   17.0.10202.2001
TypeScript Tools for Microsoft Visual Studio

Visual Basic Tools   4.2.0-1.22108.11+a29b6ae89a8885856e4c55a0e92ba056e1530458
Visual Basic components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Visual F# Tools   17.1.0-beta.22101.5+0cce0cbe615b0a944c7b95da646b0dd643f88758
Microsoft Visual F# Tools

Visual Studio Code Debug Adapter Host Package   1.0
Interop layer for hosting Visual Studio Code debug adapters in Visual Studio

Visual Studio Container Tools Extensions   1.0
View, manage, and diagnose containers within Visual Studio.

Visual Studio IntelliCode   2.2
AI-assisted development for Visual Studio.

Visual Studio Tools for CMake   1.0
Visual Studio Tools for CMake

Visual Studio Tools for Containers   1.0
Visual Studio Tools for Containers

VisualStudio.DeviceLog   1.0
Information about my package

VisualStudio.Foo   1.0
Information about my package

VisualStudio.Mac   1.0
Mac Extension for Visual Studio

VSColorOutput64   2021.3
Color output for build and debug windows - https://mike-ward.net/vscoloroutput

Workflow Manager Tools 1.0   1.0
This package contains the necessary Visual Studio integration components for Workflow Manager.

Xamarin   17.2.0.35 (main@13fdda6)
Visual Studio extension to enable development for Xamarin.iOS and Xamarin.Android.

Xamarin Designer   17.2.0.31 (remotes/origin/main@489d90aeb)
Visual Studio extension to enable Xamarin Designer tools in Visual Studio.

Xamarin Templates   17.1.13 (bb31b34)
Templates for building iOS, Android, and Windows apps with Xamarin and Xamarin.Forms.

Xamarin.Android SDK   12.2.99.75 (main/398afd4)
Xamarin.Android Reference Assemblies and MSBuild support.
    Mono: a5d1934
    Java.Interop: xamarin/java.interop/main@32635fd6
    ProGuard: Guardsquare/proguard/v7.0.1@912d149
    SQLite: xamarin/sqlite/3.37.2@51b4821
    Xamarin.Android Tools: xamarin/xamarin-android-tools/main@0e80ea1


Xamarin.iOS and Xamarin.Mac SDK   15.7.0.163 (fa0342922)
Xamarin.iOS and Xamarin.Mac Reference Assemblies and MSBuild support.

**Reply by Shyam-Gupta** [reply]
Thank you for sharing the data. I installed all the workloads and extensions (except for 2) but couldn't repro the issue.
I couldn't find following 2 extensions:

- JavaScript project system
- Workflow Manager Activities

Could you please try disabling them once and see if it resolves the designer load issue ? To disable them, select the extension in Manage Extensions dialog and click 'Disable'. Restart Visual Studio.

You can enable them again, from Manage Extensions dialog -> Select extension -> 'Enable', followed by Visual Studio restart.

**Reply by thomasmiko** [reply]
Unfortunately, after uninstalling the 2 extensions, the problem still exists.

Is there a more reliable way to somehow find out which path is the problem here (within the EmulateFileIOPermissionChecks function)?

I have adapted an example project for a .NET profiler so that I can log the functions called in devenv.exe (including the parameters). But here are actually only - from my point of view - valid paths were logged. 

Some Examples

function EmulateFileIOPermissionChecks
string length: 92
string: C:\Users\thomas.mikolajczyk\AppData\Local\Temp\VSTelem\0e3821ea139e4a53abd80977b413130729604

function EmulateFileIOPermissionChecks
string length: 108
string: C:\Users\thomas.mikolajczyk\AppData\Local\Microsoft\Windows\INetCache\IE\YJQCOG89\dyntelconfig.json[1].cache

function InternalCreateDirectoryHelper
string length: 123
string: \\?\C:\Users\thomas.mikolajczyk\AppData\Local\Microsoft\VSApplicationInsights\vstelAIF-312cbd79-9dbb-4c48-a7da-3cc2a931cb70

function EmulateFileIOPermissionChecks
string length: 123
string: \\?\C:\Users\thomas.mikolajczyk\AppData\Local\Microsoft\VSApplicationInsights\vstelAIF-312cbd79-9dbb-4c48-a7da-3cc2a931cb70

function InternalCreateDirectoryHelper
string length: 86
string: C:\Users\THOMAS~1.MIK\AppData\Local\Temp\VSTelem\289787b534894742a4c6940798233a0f33896

function EmulateFileIOPermissionChecks
string length: 92
string: C:\Users\thomas.mikolajczyk\AppData\Local\Temp\VSTelem\289787b534894742a4c6940798233a0f33896

**Reply by RussKie** [reply]
> Is it somehow possible to attach the log file (*.PML) here? Or can I paste the CSV output from the Process Monitor?

Yes, you can attach files here. You may need to zip or give a different extensions (e.g., .txt). Be mindful though that this is a public space and your logs may contain personal information (e.g., you username, etc.), consider if you are ok to share it.

Since we narrowed down the issue to the specific workstation I would try to capture the trace with procmon and see if it captures any errors. Also verify you can navigate to the `\\?\` prefixed-paths.
Some customers reported seemingly random issues with the designer were caused by their antivirus solutions, so perhaps see if you could temporarily disable it or add an exception to the AV rules.

Also could you please try starting the VS as administrator and check if you get the same issue?

**Comment by merriemcgaw** [comment]
Given that it's only on your machine, have you tried repairing or uninstall/reinstalling of VS?

**Reply by thomasmiko** [reply]
I've tried repair (didn't work) und installing an separate instance (2022 Preview), both didn't work

**Comment by Shyam-Gupta** [comment]
@thomasmiko Could you please check that by any chance following AppContextSwitchOverride is set to true in your instance of `devenv.exe.config` file: `Switch.System.IO.UseLegacyPathHandling`.

`devenv.exe.config` is usually installed at: `C:\Program Files\Microsoft Visual Studio\2022\Common7\IDE` location.
In case the switch is present, try removing it or set it to false and see if it fixes the issue. Please take a backup of the config file before making the change. Thanks.

**Reply by thomasmiko** [reply]
The UseLegacyPathHandling Switch wasn't set in the devenv.exe.config file, but i've added this key for testing:

<AppContextSwitchOverrides value="Switch.System.IO.UseLegacyPathHandling=false;" />

And now it's working and the designer for .net 6 projects will ne opened... So I think the default of this switch is true? Or can the default somehow be set differently - PC dependent?

**Reply by Shyam-Gupta** [reply]
It's great to know that after removing this switch, WinForms designer is working fine.
I don't see this switch at all in VS 2019 or 2022 `devenv.exe.config` file on my machine. So there should be no need to set it manually either. Default value of this switch is `false`.

Would you mind sharing your scenario why you need to add this switch for testing ? Thanks.

FYI: @merriemcgaw

**Reply by thomasmiko** [reply]
I think I had written my answer a bit misleading earlier.

I had no entry in the devenv.exe.config file with this switch (neither true nor false). Today i have added the switch in the devenv.exe.config (with false value), and since then the call of the designer works.

**Reply by Shyam-Gupta** [reply]
Thank you for the clarification.
I _guess_ there is some component installed against your Visual Studio instance, that is changing the value of this AppContext switch to `true` at runtime. We checked all the extensions earlier but that didn't help. Do you recall installing any other 3rd party software which might cause this to happen ?

**Reply by JeremyKuhne** [reply]
It could be set in the registry: https://docs.microsoft.com/dotnet/api/system.appcontext

Changing it programmatically isn't really plausible as it sticks with whatever value it has on first read, which for this switch is very, very early in the startup of the CLR.

**Comment by thomasmiko** [comment]
After I set the switch in the devenv.exe.config, calling the .NET 6 Designer worked. Unfortunately, I have another problem because of this.

I now have the code editor crashing when I write anything in it.
The exception is as follows:

```
Anwendung: devenv.exe
Frameworkversion: v4.0.30319
Beschreibung: Die Anwendung forderte die Beendigung des Prozesses durch System.Environment.FailFast(Zeichenfolgenmeldung) an.
Meldung: Failed to schedule time on the UI thread. A continuation would never execute.
Stapel:
   bei System.Environment.FailFast(System.String, System.Exception)
   bei Microsoft.VisualStudio.Threading.JoinableTaskFactory+MainThreadAwaiter.OnCompleted(System.Action, Boolean)
   bei System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1[[System.Threading.Tasks.VoidTaskResult, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]].AwaitUnsafeOnCompleted[[Microsoft.VisualStudio.Threading.JoinableTaskFactory+MainThreadAwaiter, Microsoft.VisualStudio.Threading, Version=17.2.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a],[Microsoft.CodeAnalysis.Editor.Tagging.AbstractAsynchronousTaggerProvider`1+TagSource+<>c__DisplayClass37_0+<<EnqueueWork>b__0>d[[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]], Microsoft.CodeAnalysis.EditorFeatures, Version=4.2.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35]](MainThreadAwaiter ByRef, <<EnqueueWork>b__0>d<System.__Canon> ByRef)
   bei Microsoft.CodeAnalysis.Editor.Tagging.AbstractAsynchronousTaggerProvider`1+TagSource+<>c__DisplayClass37_0+<<EnqueueWork>b__0>d[[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]].MoveNext()
   bei System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start[[Microsoft.CodeAnalysis.Editor.Tagging.AbstractAsynchronousTaggerProvider`1+TagSource+<>c__DisplayClass37_0+<<EnqueueWork>b__0>d[[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]], Microsoft.CodeAnalysis.EditorFeatures, Version=4.2.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35]](<<EnqueueWork>b__0>d<System.__Canon> ByRef)
   bei Microsoft.CodeAnalysis.Editor.Tagging.AbstractAsynchronousTaggerProvider`1+TagSource+<>c__DisplayClass37_0[[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]].<EnqueueWork>b__0()
   bei System.Threading.Tasks.ContinuationResultTaskFromResultTask`2[[System.Boolean, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089],[System.__Canon, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089]].InnerInvoke()
   bei System.Threading.Tasks.Task.Execute()
   bei System.Threading.ExecutionContext.RunInternal(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   bei System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   bei System.Threading.Tasks.Task.ExecuteWithThreadLocal(System.Threading.Tasks.Task ByRef)
   bei System.Threading.Tasks.Task.ExecuteEntry(Boolean)
   bei System.Threading.ThreadPoolWorkQueue.Dispatch()
```

I have now removed the switch again and since then the code editor works again, but the designer unfortunately not.
I suppose I would have to reinstall my computer to fix all these problems?

**Reply by merriemcgaw** [reply]
@RussKie could this be related to the timeout issue? Which Preview is that extension coming in? I'd really like to diagnose this if possible.

**Reply by RussKie** [reply]
> could this be related to the timeout issue

I very much doubt it could.

**Reply by RussKie** [reply]
@thomasmiko could you check if there are AppContextSwitchOverrides set in the registry? https://docs.microsoft.com/dotnet/api/system.appcontext?view=net-6.0#appcontext-for-library-consumers

Also, can you try and see if you get the same issue under a different user on the same workstation? To try to narrow down whether there's something in your profile that's causing the issue, or there's something at the global level....

**Reply by thomasmiko** [reply]
@RussKie I've checked the registry, but i have no overrides within the registry.
![appcontext overrides registry 1](https://user-images.githubusercontent.com/55090340/160430297-5eec27a5-3f72-4d67-85d1-02d1293123bd.png)
![appcontext overrides registry 2](https://user-images.githubusercontent.com/55090340/160430313-ce6c8e77-1117-40db-8641-193b0ca5207b.png)

I've also tried it with another user on my machine, but the error is the same. So it should be an workstation related problem.

**Reply by RussKie** [reply]
Thank you for checking. I'm really out of ideas, sorry.
